### DISCUSSION TITLE
Simplifiy register by requireing plugins to define --encoding flag

### DISCUSSION THREAD

**Discussion by logankaser** [nushell/nushell#5894] [Ideas]
Plugins could be required to provide a `--encoding` that prints a string `"capnp"` or `"json"` to inform nushell about which encoding they want.
Alternatively, this could be used as a fallback if the user does not pass `--encoding` to register

**Comment by kubouch** [comment]
Can you make a feature request issue for this? Sounds useful to me, but @elferherrera would know better.

**Reply by logankaser** [reply]
#5944
